### ISSUE TITLE
[Request] Anyway to use Nacon Revolution Pro Controller 2 ?

### ISSUE THREAD

**Issue by Ilitd** [Ryochan7/DS4Windows#230]
Hello,

I see that version 1 of this controller is supported by DS4W (VID 0x146B, PID 0x601) but not version 2 (same VID, PID 0x602).

(I tried to build my own DS4W by adding PID in DS4Devices.cs but didn't work for me. Well, I am a noob in programming...)

Any possibility to include this controller in the future of DS4W ?

The v2 of this controller has 3 modes : PS4 (Normal DualShock 4), PS4 Advance and PC. Of course PC mode has XInput and works flawlessly, but touchpad cannot be used unfortunately.

**Comment by Ryochan7** [comment]
Since I do not have access to the hardware, I cannot attempt to modify DS4Windows to add support for it. The most I could do is attempt to plug the VID + PID combination into the program and have someone test it to see if that is the only modification that is required. The better route would require someone with the Nacon Revolution Pro Controller 2 to modify DS4Windows to include support for it and then create a pull request so the modification can be included in the main build.

**Comment by Ryochan7** [comment]
Test build with VID + PID combination included

http://ryochan7.xyz/ds4windows/test/DS4Windows_NaconProV2_x64_20180320.zip

**Comment by Ilitd** [comment]
Hi Ryochan7,

Thank you for the test build !
I'll try it tomorrow night (EU time) as soon as I can (I don't have the controller right now).

**Comment by Ilitd** [comment]
Hello Ryohan7,

Unfortunately, the test build didn't work.
I will also try to find a way, but it could take a long time since I lack time these days.

**Comment by Yuki-nyan** [comment]
I now have one of these and will see if I can get it working on par with the previous model.

**Comment by Yuki-nyan** [comment]
My pull request was accepted so you should find that this works with the next build. I also put in a workaround for the broken gyro support on Nacon controllers.

**Comment by eedugon** [comment]
Hi @Ilitd , @Yuki-nyan ,

How have you managed to detect this controller from DS4W? I have a new one and it's not recognized at all.
What mode should I set the controller to? (PS4 / PS4 Adv / PC)?
I use the USB cable, and I've tried with different versions of DS4W and all modes.
I have tried the basic installation and also with HIDGuardian, same results... nothing detected. And device manager looks fine, the device is there `VID 0x146B, PID 0x602`

Any hint? It might be someting silly, as It's the first time I'm trying DS4Windows.

Thanks in advance!
